src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -123,6 +123,16 @@ public function compile($path = null)
             );
 
             if (! empty($this->getPath())) {
+                $tokens = $this->getOpenAndClosingPHPTokens($contents);
+
+                // If the tokens we retrieved from the compiled contents have at least
+                // one opening tag and if that last token isn't the closing tag, we
+                // need to close the statement before adding the path at the end.
+                if ($tokens->intersect([T_OPEN_TAG, T_OPEN_TAG_WITH_ECHO])->isNotEmpty()
```

Comment:
```suggestion
                if ($tokens->isNotEmpty()
```

Tests still pass without the `->intersect()`. This can probably be removed?

Reply:
You're correct. I've removed it.